webapp/src/components/sidebar_right/index.tsx

```diff
@@ -0,0 +1,255 @@
+import {useDispatch, useSelector} from 'react-redux';
+
+import {getCurrentUserLocale} from 'mattermost-redux/selectors/entities/i18n';
+import {Theme} from 'mattermost-redux/types/preferences';
+import {GlobalState} from 'mattermost-redux/types/store';
+import {makeStyleFromTheme, changeOpacity} from 'mattermost-redux/utils/theme_utils';
+
+import {getYourPrDetails, getReviewDetails} from 'src/actions';
+import {RHSStates} from 'src/constants';
+import {getPluginState} from 'src/selectors';
+import {Item} from 'src/types/gitlab_items';
+
+import I18nProvider from '../i18n_provider';
+
+import GitlabItems from './gitlab_items';
+
+const AUTO_HIDE_TIMEOUT = 500;
+const AUTO_HIDE_DURATION = 500;
+
+interface Props {
+    username: string;
+    org: string;
+    gitlabURL: string;
+    reviews: Item[];
+    unreads: Item[],
+    yourPrs: Item[],
+    yourAssignments: Item[],
+    rhsState: string,
+    theme: Theme,
+}
+
+export function renderView(props: Props) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--view'
+        />);
+}
+
+export function renderThumbHorizontal(props: Props) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--horizontal'
+        />);
+}
+
+export function renderThumbVertical(props: Props) {
+    return (
+        <div
+            {...props}
+            className='scrollbar--vertical'
+        />);
+}
+
+function mapGitlabItemListToPrList(gilist: Item[]) {
+    if (!gilist) {
+        return [];
+    }
+
+    return gilist.map((pr: Item) => {
+        return {sha: pr.sha, project_id: pr.project_id, iid: pr.iid};
+    });
+}
+
+function shouldUpdateDetails(prs: Item[], prevPrs: Item[], targetState: string, currentState: string) {
+    if (currentState !== targetState) {
+        return false;
+    }
+
+    if (prs.length !== prevPrs.length) {
+        return true;
+    }
+
+    for (let i = 0; i < prs.length; i++) {
+        if (prs[i].id !== prevPrs[i].id) {
+            return true;
+        }
+    }
+    return false;
+}
+
+function mapPrsToDetails(prs: Item[], details: Item[]) {
+    if (!prs || !prs.length) {
+        return [];
+    }
+
+    return prs.map((pr) => {
+        const foundDetails = details && details.find((prDetails) => pr.project_id === prDetails.project_id && pr.sha === prDetails.sha);
+        if (!foundDetails) {
+            return pr;
+        }
+
+        return {
+            ...pr,
+            status: foundDetails.status,
+            approvers: foundDetails.approvers,
+            total_reviewers: pr.reviewers.length,
+        };
+    });
+}
+
+function SidebarRight({theme}: {theme: Theme}) {
+    const {username, yourAssignments, org, unreads, gitlabURL, rhsState, reviews, yourPrs, reviewDetails, yourPrDetails, getUserLocale} = useSelector((state: GlobalState) => {
+        const store = getPluginState(state);
+        return {
+            username: store.username,
```

Comment:
This will recompute (and re-render) every time something in the redux store changes. Can we instead define this selector elsewhere with `createSelector`?